lib/storage-client.js

```diff
@@ -0,0 +1,455 @@
+    ? null
+    : node.firstChild.nodeValue;
+}
+
+
+class ChromedriverStorageClient {
+  constructor (args = {}) {
+    const {
+      chromedriverDir = getChromedriverDir(),
+      timeout = TIMEOUT_MS,
+    } = args;
+    this.chromedriverDir = chromedriverDir;
+    this.timeout = timeout;
+    this.mapping = {};
+  }
+
+  /**
+   * @typedef {Object} AdditionalDriverDetails
+   * @property {?string} version - Chromedriver version
+   * or `null` if it cannot be found
+   * @property {?string} minBrowserVersion - The minimum browser version
+   * supported by chromedriver or `null` if it cannot be found
+   */
+
+  /**
+   * Gets additional chromedriver details from chromedriver
+   * release notes
+   *
+   * @param {string} content - Release notes of the corresponding chromedriver
+   * @returns {AdditionalDriverDetails}
+   */
+  parseNotes (content) {
+    const result = {};
+    const versionMatch = /^\s*[-]+ChromeDriver[\D]+([\d.]+)/im.exec(content);
+    if (versionMatch) {
+      result.version = versionMatch[1];
+    }
+    const minBrowserVersionMatch = /^\s*Supports Chrome[\D]+(\d+)/im.exec(content);
+    if (minBrowserVersionMatch) {
+      result.minBrowserVersion = minBrowserVersionMatch[1];
+    }
+    return result;
+  }
+
+  /**
+   * Downloads chromedriver release notes and puts them
+   * into the dictionary argument
+   *
+   * @param {string} driverKey - Driver version plus archive name
+   * @param {string} notesUrl - The URL of chromedriver notes
+   * @param {Object} infoDict - The dictionary containing driver info.
+   * The method call mutates by merging `AdditionalDriverDetails`
+   * @throws {Error} if the release notes cannot be downloaded
+   */
+  async retrieveAdditionalDriverInfo (driverKey, notesUrl, infoDict) {
+    const response = await requestPromise({
+      url: notesUrl,
+      method: 'GET',
+      headers: {
+        'user-agent': 'appium',
+        accept: '*/*',
+      },
+      resolveWithFullResponse: true,
+      timeout: this.timeout,
+    });
+    const { minBrowserVersion } = this.parseNotes(response.body);
+    if (!minBrowserVersion) {
+      log.debug(`The driver '${driverKey}' does not contain valid notes. Skipping it`);
+      return;
+    }
+    infoDict.minBrowserVersion = minBrowserVersion;
+  }
+
+  /**
+   * Parses chromedriver storage XML and stores
+   * the parsed results into `this.mapping`
+   *
+   * @param {DOMDocument} doc - The DOM representation
+   * of the chromedriver storage XML
+   * @param {boolean} shouldParseNotes [true] - If set to `true`
+   * then additional drivers information is going to be parsed
+   * and assigned to `this.mapping`
+   */
+  async parseStorageXml (doc, shouldParseNotes = true) {
+    const driverNodes = xpath.select(`//*[local-name(.)='Contents']`, doc);
+    log.debug(`Parsed ${driverNodes.length} entries from storage XML`);
+    if (_.isEmpty(driverNodes)) {
+      return;
+    }
+
+    const fibers = [];
```

Comment:
it was confusing to me to have these called `fibers` when we're not using node-fibers. it made me think we're doing something with threads when in fact it's all still just promises.

Reply:
changed the name to `promises`